q2cli/tests/test_usage.py

```diff
@@ -0,0 +1,227 @@
+            "    --o-left left1.qza \\",
+            "    --o-right right1.qza \\",
+            "    --o-left-viz left_viz1.qzv \\",
+            "    --o-right-viz right_viz1.qzv",
+            "qiime dummy-plugin typical-pipeline \\",
+            "    --i-int-sequence left1.qza \\",
+            "    --i-mapping out_map1.qza \\",
+            "    --p-no-do-extra-thing \\",
+            "    --o-out-map out_map2.qza \\",
+            "    --o-left left2.qza \\",
+            "    --o-right right2.qza \\",
+            "    --o-left-viz left_viz2.qzv \\",
+            "    --o-right-viz right_viz2.qzv"
+        ),
+    ),
+    ###########################################################################
+
+    # identity with metadata simple example ###################################
+    (
+        'identity_with_metadata',
+        'identity_with_metadata_simple',
+        (
+            "qiime dummy-plugin identity-with-metadata \\",
+            "    --i-ints ints.qza \\",
+            "    --m-metadata-file md.tsv \\",
+            "    --o-out out.qza"
+        ),
+    ),
+    ###########################################################################
+
+    # identity with metadata merging example ##################################
+    (
+        'identity_with_metadata',
+        'identity_with_metadata_merging',
+        (
+            "qiime dummy-plugin identity-with-metadata \\",
+            "    --i-ints ints.qza \\",
+            "    --m-metadata-file md1.tsv \\",
+            "    --m-metadata-file md2.tsv \\",
+            "    --o-out out.qza"
+        ),
+    ),
+    ###########################################################################
+
+    # identity with metadata column get metadata column example ###############
+    (
+        'identity_with_metadata_column',
+        'identity_with_metadata_column_get_mdc',
+        (
+            "qiime dummy-plugin identity-with-metadata-column \\",
+            "    --i-ints ints.qza \\",
+            "    --m-metadata-file md.tsv \\",
+            "    --m-metadata-column 'a' \\",
+            "    --o-out out.qza"
+        )
+
+    ),
+    ###########################################################################
+
+
+    # optional inputs example #################################################
+    (
+        'optional_artifacts_method',
+        'optional_inputs',
+        (
+            "qiime dummy-plugin optional-artifacts-method \\",
+            "    --i-ints ints.qza \\",
+            "    --p-num1 1 \\",
+            "    --o-output output.qza",
+            "qiime dummy-plugin optional-artifacts-method \\",
+            "    --i-ints ints.qza \\",
+            "    --p-num1 1 \\",
+            "    --p-num2 2 \\",
+            "    --o-output output.qza",
+            "qiime dummy-plugin optional-artifacts-method \\",
+            "    --i-ints ints.qza \\",
+            "    --p-num1 1 \\",
+            "    --p-num2 None \\",
```

Comment:
@thermokarst do we actually want to pass `None` here?

Reply:
Not if it wasn't explicitly set in the usage example.